platforms/documentation/docs/src/docs/userguide/best-practices/best_practices_general.adoc

```diff
@@ -175,3 +175,119 @@ include::sample[dir="snippets/bestPractices/avoidInternal/groovy",files="build.g
 === Tags
 
 <<tags_reference.adoc#tag:upgrades,`#upgrades`>>
+
+[[no_source_in_root]]
+== Do Not Put Source Files in the Root Project
+:keywords: structuring-builds
+
+Do not put source files in your root project; instead, put them in an included (sub)project.
+
+=== Explanation
+
+The root project is a special project in Gradle that serves as the entry point for your build.
+
+It is a place to configure the build as a whole, such as declaring (but not applying) plugins to ensure the same version is available in every project in a build, and to do other types of configuration that apply to all the build's projects.
+
+It should not be used for source files, instead they should be located in a separate (sub)project.
+
+=== Example
+
+==== Don't Do This
+
+[.multi-language-sample]
+=====
+.A common way to structure new builds
+[source, kotlin]
+----
+├── app
+├── build.gradle.kts
```

Comment:
Should we note this as optional or remove it?

Reply:
A buildscript is necessary to apply the `java-library` plugin in this project.  I'll add a comment.  

I also notice that `/app` is in the tree is a result of a manual editing mistake that I'll fix now.